tokens/cnft-vault/programs/cnft-vault/src/lib.rs

```diff
@@ -0,0 +1,295 @@
+use anchor_lang::prelude::*;
+use solana_program::{pubkey::Pubkey};
+use spl_account_compression::{
+    program::SplAccountCompression, Noop,
+};
+use mpl_bubblegum::{state::TreeConfig};
+
+declare_id!("CNftyK7T8udPwYRzZUMWzbh79rKrz9a5GwV2wv7iEHpk");
+
+#[derive(Clone)]
+pub struct MplBubblegum;
+
+impl anchor_lang::Id for MplBubblegum {
+    fn id() -> Pubkey {
+        mpl_bubblegum::id()
+    }
+}
+
+#[program]
+pub mod cnft_vault {
+
+    use super::*;
+
+    pub fn withdraw_cnft<'info>(ctx: Context<'_, '_, '_, 'info, Withdraw<'info>>,
+        root: [u8; 32],
+        data_hash: [u8; 32],
+        creator_hash: [u8; 32],
+        nonce: u64,
+        index: u32,) -> Result<()> {
+        msg!("attempting to send nft {} from tree {}", index, ctx.accounts.merkle_tree.key());
+
+        // CPI to bubblegum
```

Comment:
Commented code still needed?

Reply:
Agreed please remove

Reply:
maybe someone figures out a way how to do it with Anker..
and okay, i can remove it.